tests/typing/bad/poly_eq.catala_en

```diff
@@ -0,0 +1,28 @@
+```catala-metadata
+declaration find_in_list
+  content optional of anything of type t
+  depends on
+    needle content anything of type t,
+    haystack content list of anything of type t
+  equals
+    combine all elt among haystack
+      in found initially Absent
+      with
+        if elt = needle then Present content elt
+        else found
+```
+
+```catala-test-cli
+$ catala typecheck
+┌─[ERROR]─
+│
+│  Equality cannot be resolved at this point: the type of the operands is not
+│  fully known.
```

Comment:
I would have put : "The left and right hand side of this equality are polymorphic and can have any type. Hence, we cannot test if they're equal since we don't know their shape in advance. This comparison is thus forbidden."

Reply:
Won't users be more interested in something actionable rather than a more technical explanation ? I was actually hesitant to write something along the lines of "You will need to write specific functions for the different concrete types that you need to handle at this point"

We should be able to lift this limitation in the future when we reenable the monomorphisation pass ; but at the moment, without this check, it would pass in the interpreter but fail once compiled.